File: lib/src/main/java/uk/iatom/rfm69/registers/OpModes.java
package uk.iatom.rfm69.magicbytes;

public enum OpModes {
    
    // RegOpMode
    SEQUENCER_OFF(0x80),
    SEQUENCER_ON(0x00), // Default

    LISTEN_ON(0x40),
    LISTEN_OFF(0x00), // Default

    LISTENABORT(0x20),
    SLEEP(0x00),
    STANDBY(0x04), // Default
    SYNTHESIZER(0x08),
    TRANSMITTER(0x0C),
    RECEIVER(0x10);

    private byte val;
    
    OpModes(int val) {
        this.val = (byte) val;
    }

    public byte getVal() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/register/values/OpModes.kt
package uk.iatom.rfm69.register.values

import uk.iatom.rfm69.register.ICodec
import uk.iatom.rfm69.register.IRegisterValue

// RegOpMode
class OpModes(
        var sequencer: Sequencer = Sequencer.ON,
        var listen: Listen = Listen.OFF,
        var power: Power = Power.STANDBY
             ): IRegisterValue {


    enum class Sequencer(val v: Int) {
        ON(0x80),
        OFF(0x00)
    }

    enum class Listen(val v: Int) {
        ON(0x40),
        OFF(0x00),
    }

    enum class ListenAbort(val v: Int) {
        // Default unknown //TODO Find listenabort default
        ON(0x20),
        OFF(0x00),
    }

    enum class Power(val v: Int) {
        RECEIVER(0x10),
        SYNTHESIZER(0x08),
        STANDBY(0x04),
        TRANSMITTER(0x0C),
        SLEEP(0x00),
    }

    override fun getByte(): Byte {
        return (sequencer.v or listen.v or power.v).toByte()
    }

    class Codec: ICodec<OpModes> {


        override fun encode(value: OpModes): Byte = value.getByte()

        override fun decode(byte: Byte): OpModes {
            TODO("Not yet implemented")
        }
    }
}
